src/app.rs

```diff
@@ -0,0 +1,296 @@
+
+        for (pattern, expected_branch_protection_rule) in &ctx.config.branch_protection_rules {
+            // info!(
+            //     "For pattern {} we expect {:?}",
+            //     pattern, expected_branch_protection_rule
+            // );
+
+            if let Some(actual_branch_protection_rule) =
+                actual_branch_protection_rules.remove(pattern)
+            {
+                if let Some(diff) =
+                    expected_branch_protection_rule.diff(&actual_branch_protection_rule)
+                {
+                    let branch_protection_name = actual_branch_protection_rule.pattern.as_str();
+                    debug!("Repository {} Diff: {:?}", repo_with_owner, diff);
+                    let patch = diff.dump_patch(&actual_branch_protection_rule);
+                    debug!("Repository {} Patch: {:?}", repo_with_owner, patch);
+                    if !ctx.options.dry_run_bpr {
+                        let response = ctx.api_client.update_branch_protection(
+                            repo_owner,
+                            repo_name,
+                            branch_protection_name,
+                            patch,
+                        )?;
+
+                        debug!("Response: {:?}", response);
+                        debug!("Response body: {:?}", response.text()?);
+                    } else {
+                        debug!(
+                            "DRY RUN: Update branch protection rule for {}",
+                            repo_with_owner
+                        );
+                    }
+                    // Repo has a branch protection rule with this key, see if the values match!
+                } else {
+                    debug!(
+                        "Repository {} has a 100% matching branch protection rule:\n{:?}\n{:?}",
+                        repo_with_owner,
+                        actual_branch_protection_rule,
+                        expected_branch_protection_rule
+                    );
+                }
+            } else {
+                // Repo does *not* have a branch protection rule with this key, should we create it?
+                // Maybe ask the user?
+            }
+        }
+        */
+    }
+
+    fn check_topics(&mut self, ctx: &mut Context) -> anyhow::Result<()> {
+        if ctx.config.topics.is_empty() {
+            return Ok(());
+        }
+
+        let mut final_topics = self.topics.names.clone();
+
+        for operation in &ctx.config.topics {
+            match operation {
+                TopicOperation::MustExist { name } => {
+                    final_topics.insert(name.clone());
+                }
+                TopicOperation::MustNotExist { name } => {
+                    final_topics.remove(name);
+                }
+                TopicOperation::Rename { old_name, name } => {
+                    final_topics.remove(old_name);
+                    final_topics.insert(name.clone());
+                }
```

Comment:
This would insert `name` unconditionally, shouldn't it only be added if `old_name` was present and removed?

```suggestion
                TopicOperation::Rename { old_name, name } => {
                    if final_topics.remove(old_name) {
                        final_topics.insert(name.clone());
                    }
                }
```